### PR TITLE
By/allow writing pre commit output to files

### PR DESCRIPTION
**Pull request by barryyosi-panw**
## Related Issues
fixes: Allow writing pre-commit results to files in order to support `ValidateContent` script refactor ([CIAC-11199](https://jira-dc.paloaltonetworks.com/browse/CIAC-11199))

### CODE REVIEW

File: demisto_sdk/commands/pre_commit/pre_commit_command.py [hunk 1]
```diff
@@ -48,6 +47,8 @@
     PreCommitContext,
 )
 
+json = JSON_Handler()
```
Comment (thefrieddan1) [inline demisto_sdk/commands/pre_commit/pre_commit_command.py:50]: does it have to be global?

**Review by thefrieddan1 — APPROVED** [review]

File: demisto_sdk/commands/validate/default_config.toml [hunk 2]
```diff
@@ -155,6 +155,7 @@ select = [
   "IF104",
   "IF105",
   "IF106",
+  "IM100",
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/default_config.toml:158]: remove

File: demisto_sdk/commands/pre_commit/tests/pre_commit_test.py [hunk 3]
```diff
@@ -801,3 +802,38 @@ def test_system_hooks():
...
+    )
+
+    exit_code = pre_commit_manager(
+        input_files=[test_integration_path],
+        run_hook="check-ast",
+        json_output_path=tmp_path,
+    )
+    hook_output_path = tmp_path / "check-ast.json"
+    assert exit_code != 0
+    assert hook_output_path.exists()
```
Comment (YuvHayun) [inline demisto_sdk/commands/pre_commit/tests/pre_commit_test.py:839]: let's also validate the file content.

File: .changelog/4696.yml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+changes:
+- description: YmlSplitter fix.
```
Comment (YuvHayun) [inline .changelog/4696.yml:2]: Fixed an issue where YmlSplitter...(complete.)

File: .changelog/4697.yml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+changes:
+- description: Allow writing pre-commit results to files.
```
Comment (YuvHayun) [inline .changelog/4697.yml:2]:
```suggestion
- description: Added support for writing pre-commit results to files.
```

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Nice job!
see my comments.

File: .changelog/4696.yml (hunk 4 above)
Comment (barryyosi-panw) [reply .changelog/4696.yml:2]: Only fixing the pr number, the changelog description was already adapted in the released changelog.

File: demisto_sdk/commands/pre_commit/tests/pre_commit_test.py (hunk 3 above)
Comment (barryyosi-panw) [reply demisto_sdk/commands/pre_commit/tests/pre_commit_test.py:840]: done

File: demisto_sdk/commands/validate/default_config.toml (hunk 2 above)
Comment (barryyosi-panw) [reply demisto_sdk/commands/validate/default_config.toml]: removed

**Review by YuvHayun — APPROVED** [review]

**Pull request merged**